RoosterPlanner.Api/Controllers/ShiftController.cs

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+
+        [HttpGet("shift/{id}")]
+        public async Task<ActionResult> GetShift(Guid id)
+        {
+            if (id == Guid.Empty)
+                return BadRequest();
+            try
+            {
+                TaskResult<Shift> result = await shiftService.GetShift(id);
+                if (!result.Succeeded)
+                    return UnprocessableEntity();
+                if (result.Data == null)
+                    return Ok();
+                return Ok(ShiftViewModel.CreateVm(result.Data));
+            }
+            catch (Exception ex)
+            {
+                logger.Log(LogLevel.Error, ex.ToString());
+                Response.Headers.Add("message", ex.Message);
+                return UnprocessableEntity();
+            }
+        }
+
+        [Authorize(Policy = "Boardmember&Committeemember")]
+        [HttpPost]
+        public async Task<ActionResult> Save(List<ShiftViewModel> shiftViewModels)
+        {
+            if (shiftViewModels == null || shiftViewModels.Count == 0)
+                return BadRequest("No valid shifts received");
+
+            try
+            {
+                List<Shift> shifts = shiftViewModels.Select(ShiftViewModel.CreateShift).ToList();
+                string oid = IdentityHelper.GetOid(HttpContext.User.Identity as ClaimsIdentity);
+
+                //get project and get task from db
+                Project project = projectService.GetProjectDetails(shifts[0].ProjectId).Result.Data;
+                Task task = null;
+                if (shifts[0].TaskId != null)
+                    task = taskService.GetTask((Guid) shifts[0].TaskId).Result.Data;
+
+                foreach (Shift shift in shifts)
+                {
+                    if (shift.Id != Guid.Empty || shift.TaskId == null || shift.ProjectId == Guid.Empty)
+                        shifts.Remove(shift);
+                    
+                    // check if projectId and taskId differs from above? getproject/task => add project and task to shift
+                    if(project.Id!=shift.ProjectId)
+                        project=projectService.GetProjectDetails(shift.ProjectId).Result.Data;
+                    
+                    if(task !=null && shift.TaskId!=null &&  task.Id!=shift.TaskId)
+                        task = taskService.GetTask((Guid) shift.TaskId).Result.Data;
+
+                    if (project == null || task == null)
+                        shifts.Remove(shift);
+                    shift.Project = project;
+                    shift.Task = task;
+                    shift.LastEditDate = DateTime.UtcNow;
+                    shift.LastEditBy = oid;
+                }
+
+                if (shifts.Count() != shiftViewModels.Count())
+                    return UnprocessableEntity("Could not covert al viewmodels to shifts");
+
+                TaskListResult<Shift> result = await shiftService.CreateShifts(shifts);
+
+                if (result.Succeeded)
```

Comment:
Draai if statement om. De normale flow zou moeten eindigen in het eind van de functie